src/constructs/core/stack.ts

```diff
@@ -120,4 +122,26 @@ export class GuStack extends Stack implements StackStageIdentity, GuMigratingSta
     this.addTag("Stack", this.stack);
     this.addTag("Stage", this.stage);
   }
+
+  protected prepare(): void {
+    super.prepare();
+
+    /*
+    Log a message whenever a stateful resource is encountered in the stack.
+
+    Ideally we'd add a stack policy to stop the resource being deleted,
+    however this isn't currently supported in CDK.
+
+    See:
+      - https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/protect-stack-resources.html
+      - https://github.com/aws/aws-cdk-rfcs/issues/72
+     */
+    this.node.findAll().forEach((construct: IConstruct) => {
+      if (CfnResource.isCfnResource(construct) && StatefulResourceTypes.includes(construct.cfnResourceType)) {
+        Logger.warn(
```

Comment:
We could call [`addDeletionOverride`](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_core.CfnResource.html#addwbrdeletionwbroverridepath) here, as we do for the [`GuAutoScalingGroup`](https://github.com/guardian/cdk/blob/31cea549bcf9a1a590bb217a3967c660c4084bf3/src/constructs/autoscaling/asg.ts#L143).